diff -Nru icradius-0.17b.orig/src/Make.inc icradius-0.17b/src/Make.inc
--- icradius-0.17b.orig/src/Make.inc	Sat May 19 21:36:25 2001
+++ icradius-0.17b/src/Make.inc	Sat May 19 21:36:45 2001
@@ -4,7 +4,7 @@
 #
 #
 
-RADIUS_DIR = /home/radius/etc/raddb
+RADIUS_DIR = /etc/raddb
 
 SERVER_OBJS    = radiusd.o dict.o files.o filters.o util.o md5.o attrprint.o \
 			acct.o radius.o pam.o log.o version.o proxy.o \
